client/state/google-apps-users/reducer.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * External Dependencies
+ */
+
+import { combineReducers } from 'redux';
+import uniqBy from 'lodash/uniqBy';
+import property from 'lodash/property';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	GOOGLE_APPS_USERS_FETCH,
+	GOOGLE_APPS_USERS_FETCH_COMPLETED,
+	GOOGLE_APPS_USERS_FETCH_FAILED
+} from 'state/action-types';
+
+export function items( state = [], action ) {
+	switch ( action.type ) {
+		case GOOGLE_APPS_USERS_FETCH_COMPLETED:
+			return uniqBy( state.concat( action.items ), property( 'email' ) );
+			break;
```

Comment:
`persist-redux` was just turned on for dev, so we'll need to worry about this automatically persisting. For now, lets add 2 extra cases:
```javascript
case SERIALIZE:
    return [];
case DESERIALIZE:
    return [];
```
To not clutter this PR, we can spin up a future one to add a schema and tests around it. 

For reasoning and examples see: https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#data-persistence--2754-